File: samples/clientpetstorecsharp/src/main/csharp/io/swagger/Api/StoreApi.cs
using System;
using System.Collections.Generic;
using io.swagger.client;
using io.swagger.Model;





namespace io.swagger.Api {
  
  public class StoreApi {
    private readonly ApiInvoker _apiInvoker = ApiInvoker.GetInstance();

    public StoreApi(String basePath = "http://petstore.swagger.io/v2")
    {
      BasePath = basePath;
    }

    public ApiInvoker GetInvoker() {
      return _apiInvoker;
    }

    public string BasePath { get; set; }

    

    /// <summary>
    /// Returns pet inventories by status Returns a map of status codes to quantities
    /// </summary>
    
    /// <returns></returns>
    public Dictionary<String, int?>  getInventory () {
      // create path and map variables
      var path = "/store/inventory".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(Dictionary<String, int?>) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as Dictionary<String, int?>;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          if (response != null){
             return (Dictionary<String, int?>) ApiInvoker.Deserialize(response, typeof(Dictionary<String, int?>));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Place an order for a pet 
    /// </summary>
    /// <param name="Body">order placed for purchasing the pet</param>
    
    /// <returns></returns>
    public Order  placeOrder (Order Body) {
      // create path and map variables
      var path = "/store/order".Replace("{format}","json");

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(Order) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as Order;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "POST", queryParams, Body, headerParams, formParams);
          if (response != null){
             return (Order) ApiInvoker.Deserialize(response, typeof(Order));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Find purchase order by ID For valid response try integer IDs with value &lt;= 5 or &gt; 10. Other values will generated exceptions
    /// </summary>
    /// <param name="OrderId">ID of pet that needs to be fetched</param>
    
    /// <returns></returns>
    public Order  getOrderById (string OrderId) {
      // create path and map variables
      var path = "/store/order/{orderId}".Replace("{format}","json").Replace("{" + "orderId" + "}", _apiInvoker.EscapeString(OrderId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(Order) == typeof(byte[])) {
          
          var response = _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return ((object)response) as Order;
          
          
        } else {
          
          var response = _apiInvoker.InvokeApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          if (response != null){
             return (Order) ApiInvoker.Deserialize(response, typeof(Order));
          }
          else {
            return null;
          }
          
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return null;
        }
        else {
          throw;
        }
      }
    }
    

    /// <summary>
    /// Delete purchase order by ID For valid response try integer IDs with value &lt; 1000. Anything above 1000 or nonintegers will generate API errors
    /// </summary>
    /// <param name="OrderId">ID of the order that needs to be deleted</param>
    
    /// <returns></returns>
    public void  deleteOrder (string OrderId) {
      // create path and map variables
      var path = "/store/order/{orderId}".Replace("{format}","json").Replace("{" + "orderId" + "}", _apiInvoker.EscapeString(OrderId.ToString()));

      // query params
      var queryParams = new Dictionary<String, String>();
      var headerParams = new Dictionary<String, String>();
      var formParams = new Dictionary<String, object>();

      

      

      

      

      try {
        if (typeof(void) == typeof(byte[])) {
          
          
          _apiInvoker.InvokeBinaryApi(BasePath, path, "GET", queryParams, null, headerParams, formParams);
          return;
          
        } else {
          
          
          _apiInvoker.InvokeApi(BasePath, path, "DELETE", queryParams, null, headerParams, formParams);
          return;
          
        }
      } catch (ApiException ex) {
        if(ex.ErrorCode == 404) {
          return ;
        }
        else {
          throw;
        }
      }
    }
    
  }
  
}

File: samples/server/petstore/sinatra/api/store_api.rb
require 'json'


<<<<<<< HEAD
MyApp.add_route('DELETE', '/v2/stores/order/{orderId}', {
=======
MyApp.add_route('DELETE', '/v2/store/order/{orderId}', {
>>>>>>> upstream/master
  "resourcePath" => "/Store",
  "summary" => "Delete purchase order by ID",
  "nickname" => "delete_order", 
  "responseClass" => "void", 
<<<<<<< HEAD
  "endpoint" => "/stores/order/{orderId}", 
  "notes" => "For valid response try integer IDs with value < 1000. Anything above 1000 or nonintegers will generate API errors",
  "parameters" => [
    
    
    {
      "name" => "order_id",
      "description" => "ID of the order that needs to be deleted",
      "dataType" => "string",
      "paramType" => "path",
    },
    
    
    
=======
  "endpoint" => "/store/order/{orderId}", 
  "notes" => "For valid response try integer IDs with value < 1000. Anything above 1000 or nonintegers will generate API errors",
  "parameters" => [
    {
      "name" => "order_id",
      "description" => "ID of the order that needs to be deleted",
      "dataType" => "string",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/v2/store/inventory', {
  "resourcePath" => "/Store",
  "summary" => "Returns pet inventories by status",
  "nickname" => "get_inventory", 
  "responseClass" => "map[string,int]", 
  "endpoint" => "/store/inventory", 
  "notes" => "Returns a map of status codes to quantities",
  "parameters" => [
>>>>>>> upstream/master
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/v2/stores/order/{orderId}', {
  "resourcePath" => "/Store",
  "summary" => "Find purchase order by ID",
  "nickname" => "get_order_by_id", 
  "responseClass" => "Order", 
  "endpoint" => "/stores/order/{orderId}", 
  "notes" => "For valid response try integer IDs with value <= 5 or > 10. Other values will generated exceptions",
  "parameters" => [
    {
      "name" => "order_id",
      "description" => "ID of pet that needs to be fetched",
      "dataType" => "int",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


<<<<<<< HEAD
MyApp.add_route('POST', '/v2/stores/order', {
=======
MyApp.add_route('POST', '/v2/store/order', {
>>>>>>> upstream/master
  "resourcePath" => "/Store",
  "summary" => "Place an order for a pet",
  "nickname" => "place_order", 
  "responseClass" => "Order", 
<<<<<<< HEAD
  "endpoint" => "/stores/order", 
  "notes" => "",
  "parameters" => [
    
    
    
    
=======
  "endpoint" => "/store/order", 
  "notes" => "",
  "parameters" => [
>>>>>>> upstream/master
    {
      "name" => "body",
      "description" => "order placed for purchasing the pet",
      "dataType" => "Order",
      "paramType" => "body",
    }
<<<<<<< HEAD
    
=======
>>>>>>> upstream/master
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

